src/app/store/app.effects.ts

```diff
@@ -0,0 +1,18 @@
+import { Effect } from '@ngrx/effects';
+import { map } from 'rxjs/operators';
+import * as MessageActions from '../modules/chat/store/chat.actions';
+import { Injectable } from '@angular/core';
+import { Socket } from 'ngx-socket-io';
+
+@Injectable()
+export class AppEffects {
+
+  constructor(private socket: Socket) {}
+
+  @Effect()
+  takeSocketEvent$ = this.socket.fromEvent('chat').pipe(
+    map((res: {action: string, payload?: object}) => {
+      return new MessageActions.ChatUpdated(res);
```

Comment:
Fix your WS actions structure. It should be `{type: string, payload?: object}`
Do not dispatch this specific action here. Use:
`of({ type, payload }))`

Reply:
@myownsumm Do you mean something like this:
![Screen Shot 2019-11-27 at 3 16 34 PM](https://user-images.githubusercontent.com/23140788/69726320-f41b1100-1128-11ea-90af-6e6dc00592b4.png)

Reply:
Fixed